prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```
@@ -130,11 +133,19 @@ class WorkflowEngine(
             val header = createHeader(task, reportFile, itemLineages, organization, receiver)
             val currentEventAction = Event.EventAction.parseQueueMessage(header.task.nextAction.literal)
             // Ignore messages that are not consistent with the current header
-            if (currentEventAction != messageEvent.eventAction) return@transact
+            if (currentEventAction != messageEvent.eventAction) {
+                context?.let {
+                    context.logger.warning(
+                        "Weird error for $reportId: queue event = ${messageEvent.eventAction.name}, " +
+                            " but task.nextAction = ${currentEventAction.name} "
+                    )
+                }
+                return@transact
+            }
```

This was a silent failure.